### PR TITLE
Move remaining lobby server code to lobby project

### PR DESCRIPTION
**Pull request by ssoloff**
## Overview

Moves the remaining lobby server code from the `game-core` project to the `lobby` project.  This should be the last PR related to moving lobby-specific code to the `lobby` project.

## Functional Changes

None.

## Refactoring Changes

* When I moved `LobbyPropertyReader`, I changed its package from `g.s.engine.config.lobby` to `g.s.engine.lobby.server.config` to be more consistent with the names of the other packages in the `lobby` project.
* The `g.s.test.Assertions` class was moved to `o.t.test.common` in the `test-common` project so it can be used by both the `game-core` and `lobby` projects.

## Manual Testing Performed

Smoke tested the `lobby` project build artifact to ensure I could:

* Connect a lobby client
* Connect a headless game server
* Join a headless game server
* Mute lobby users

### CODE REVIEW

File: lobby/build.gradle [hunk 1]
```diff
@@ -15,6 +15,9 @@ jar {
 
 dependencies {
     compile project(':game-core')
+    compile 'org.mindrot:jbcrypt:0.4'
+
+    runtime "org.postgresql:postgresql:$postgresqlVersion"
```
Comment (RoiEXLab) [inline lobby/build.gradle]: It's so nice to see this 👍

**Comment by ssoloff** [comment]
> Unfortunately I'm convinced that decoupling game-core from the UI is going to be a lot harder, not quite a hard as decoupling game-core from the headless game server code though.

Agreed.  Within the next week, I was planning on submitting a preview PR to add new modules for the headless and headed game runners.  It wouldn't really change anything for now; just give us a starting point to begin splitting things out of `game-core`.

> we should change the install4j deployment to only deploy the game-core project for now, so that it shrinks in size.

I'm assuming you're talking about when splitting off the headless game server and not the lobby.  The code in the `lobby` project is already excluded from the installers.  Here are the artifact sizes before and after this PR:

Artifact | Size Before (MiB) | Size After (MiB)
:-- | :-- | :--
triplea-1.9.0.0.11540-all_platforms.zip | 25.1 | 24.4
triplea-1.9.0.0.11540-server.zip | 17.9 | 17.9
TripleA_1.9.0.0.11540_macos.dmg | 15.5 | 15.3
TripleA_1.9.0.0.11540_unix.sh | 15.3 | 15.0
TripleA_1.9.0.0.11540_windows-32bit.exe | 16.4 | 16.2
TripleA_1.9.0.0.11540_windows-64bit.exe | 16.5 | 16.3

> We also want to tell @prastle when changing this, because he might need to modify his scripts accordingly to host games.

Again, assuming you're talking about the headless game server here.  Agreed.  I planned on CCing @prastle when I submit the aforementioned PR to get his feedback on moving the headless game server to a separate artifact.

> do you think there'd be a downside if we no longer used fat jars for deployment with install4j and just dumped all dependency jars into the bin folder?

Not really sure.  I don't think we have users that still replace the fat jar as an "upgrade," and that was a big motivator behind having a single jar.

**Comment by RoiEXLab** [comment]
@ssoloff 
>  The code in the lobby project is already excluded from the installers. Here are the artifact sizes before and after this PR:

Interesting, the last time I checked a triplea-all file was inside the bin directory, not sure if this changed since then.
In case it didn't change, the -all suffix is kinda confusing as it implies that everything is packed into the jar.

**Comment by ssoloff** [comment]
I believe _-all_ is the standard suffix used by the `shadowJar` task to indicate a fat jar.  See [here](http://imperceptiblethoughts.com/shadow/#configuring_output_name).

**Pull request merged**